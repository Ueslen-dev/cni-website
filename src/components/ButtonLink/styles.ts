import styled from 'styled-components';

export const Wrapper = styled.div`
  a {
    background-color: white;
    padding: 1rem 3rem 1rem;
    border-radius: 0.5rem;
    font-size: 1.6rem;
  }
`;
