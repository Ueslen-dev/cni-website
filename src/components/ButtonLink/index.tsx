import React from 'react';
import Link from 'next/link';
import * as S from './styles';

type Props = {
  label: string;
  path: string;
};

const ButtonLink = ({ label, path }: Props) => (
  <S.Wrapper>
    <Link href={path}>
      <a>{label}</a>
    </Link>
  </S.Wrapper>
);
export default ButtonLink;
