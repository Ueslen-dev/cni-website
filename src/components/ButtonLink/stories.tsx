import { Story, Meta } from '@storybook/react/types-6-0';

import ButtonLink from '.';

export default {
  title: 'ButtonLink',
  component: ButtonLink,
  args: {
    label: 'Acessar portal',
    path: '/'
  }
} as Meta;

export const Basic: Story = (args) => (
  <ButtonLink label="Acessar portal" path="/" {...args} />
);
