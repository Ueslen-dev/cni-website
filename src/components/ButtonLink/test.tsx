import { render } from '@testing-library/react';

import ButtonLink from '.';

describe('<Button/>', () => {
  it('should render the heading', () => {
    const { container } = render(
      <ButtonLink label="Acessar portal" path="/" />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
