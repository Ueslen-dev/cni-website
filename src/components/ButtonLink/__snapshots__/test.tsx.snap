// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render the heading 1`] = `
.c0 a {
  background-color: white;
  padding: 1rem 3rem 1rem;
  border-radius: 0.5rem;
  font-size: 1.6rem;
}

<main
  class="c0"
>
  <a
    href="/"
  >
    Acessar portal
  </a>
</main>
`;
